move/its/sources/service.move

```diff
@@ -273,20 +286,37 @@ module its::service {
     /// Decode an approved call and check that the source chain is trusted.
     fun decode_approved_message(self: &mut ITS, approved_message: ApprovedMessage): (String, vector<u8>) {
         let (
-            source_chain,
+            mut source_chain,
             _,
             source_address,
-            payload
+            mut payload
         ) = self.channel_mut().consume_approved_message(approved_message);
 
         assert!(self.is_trusted_address(source_chain, source_address), EUntrustedAddress);
 
+        if(source_chain.into_bytes() == AXELAR_CHAIN_NAME) {
+            let mut reader = abi::new_reader(payload);
+            if(reader.read_u256() == MESSAGE_TYPE_RECEIVE_FROM_HUB) {
```

Comment:
logic doesn't match EVM: https://github.com/axelarnetwork/interchain-token-service/pull/265/files#diff-0617e1ea35b71e4e1254741ae6248a79b107ff70dacc130cdedbbf3247daeb51R923

check if message type is receive from hub first. Then assert that source_chain is Axelarnet

Reply:
I think we need to change the EVM reference slightly. If you're sending/receiving ITS transfer to/from Axelarnet itself, I think we should still wrap/unwrap via the ITS hub message type to make the hub payloads easier to reason about. Let's change the logic to, if the trusted address is set to `"hub"`, then set destination chain to `Axelarnet`, and get its-hub address via `trustedAddress["its-hub"]` so `Axelarnet` trusted address can be set to `"hub"`